File: src/priv/bajdcc/LALR1/grammar/codegen/ICodegenBlock.java
package priv.bajdcc.LALR1.grammar.codegen;

import priv.bajdcc.LALR1.grammar.runtime.RuntimeInstUnary;

/**
 * 【目标代码生成】块接口
 *
 * @author bajdcc
 */
public interface ICodegenBlock {

	RuntimeInstUnary genBreak();
	RuntimeInstUnary genContinue();
	void enterBlockEntry(CodegenBlock block);
	void leaveBlockEntry();
	boolean isInBlock();
}

File: src/main/java/com/bajdcc/LALR1/grammar/codegen/ICodegenBlock.kt
package com.bajdcc.LALR1.grammar.codegen

import com.bajdcc.LALR1.grammar.runtime.RuntimeInstUnary

/**
 * 【目标代码生成】块接口
 *
 * @author bajdcc
 */
interface ICodegenBlock {

    val isInBlock: Boolean

    fun genBreak(): RuntimeInstUnary

    fun genContinue(): RuntimeInstUnary

    fun enterBlockEntry(block: CodegenBlock)

    fun leaveBlockEntry()
}
